hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/message/ErrorResponse.java

```diff
@@ -2,27 +2,34 @@
 
 import java.util.Optional;
 
+import org.apache.commons.lang3.StringUtils;
+
 import ca.bc.gov.hlth.hnsecure.parsing.Util;
 
 public class ErrorResponse extends ResponseSegment {
-	
+
 	private static final String segmentIdentifier = "MSA";
-		
+
 	private static final String ackKnowledgementCode = "AR";
 
 	@Override
 	public String constructResponse(HL7Message messageObj, ErrorMessage errorMessage) {
 		return constructMSH(messageObj) + constructMSA(messageObj.getMessageControlId(), errorMessage);
 	}
-	
+
 	public String constructMSA(String messageControlID, ErrorMessage errorMessage) {
 		StringBuilder sb = new StringBuilder(segmentIdentifier);
 		sb.append(Util.HL7_DELIMITER);
 		sb.append(ackKnowledgementCode);
 		sb.append(Util.HL7_DELIMITER);
 		sb.append(Optional.ofNullable(messageControlID).orElse(""));
 		sb.append(Util.HL7_DELIMITER);
-		sb.append(errorMessage.getErrorSequence() + "  " + errorMessage.getErrorMessage());
+		if (StringUtils.isNotBlank(errorMessage.getFieldValue())) {
```

Comment:
I'm wondering if the simplest approach is to have a private method (e.g. generateErrorText(HL7Message messageObj, ErrorMessage errorMessage)) where we can check the ErrorMessage enum in a switch block and add the relevant data from the HL7Message (if it's required...currently in 2 places). We could call this in constructResponse() and it would return a String containing the formatted error response which could be passed into constructMSA.